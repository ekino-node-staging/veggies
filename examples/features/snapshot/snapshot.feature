@snapshot
Feature: Using snapshot definitions

    Scenario: Snapshot testing on an api
        Given I mock http call to forward request body for path /users/yaml
        And set request json body from file
        When I POST http://fake.io/users/yaml
        Then response body should match snapshot

    Scenario: Snapshot testing on cli
        When I run command echo test
        Then exit code should be 0
        And stdout output should match snapshot

    Scenario: Snapshot testing on files
        Given I set cwd to examples/features/snapshot/fixtures
        Then file file.yaml should match snapshot

    Scenario Outline: I get all posts filtered using '<filter>=<value>' directive
        Given I mock http call to forward request body for path /users/yaml
        When I POST http://fake.io/users/yaml
        Then response status code should be 200
        And response body should match snapshot

        Examples:
            | filter       | value          |

            | title        | car industry   |

            | anotherTitle | antother value |
            # this comment and the empty lines above and below shouldn't be taken into account


            | tester       | tesfdkngjsfk   |
    Scenario: Snapshot testing on files
        Given I set cwd to examples/features/snapshot/fixtures
        Then file file.yaml should match snapshot
