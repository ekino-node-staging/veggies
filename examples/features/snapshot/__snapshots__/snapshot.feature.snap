

exports[`Scenario Outline 1.1`] = `undefined`;

exports[`Scenario Outline 2.1`] = `undefined`;

exports[`Scenario Outline 3.1`] = `undefined`;

exports[`Scenario Outline 4.1`] = `undefined`;

exports[`Setting json body from .yaml fixture file 1.1`] = `Object {
  "first_name": "Raphaël",
  "gender": "male",
  "id": 1,
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on an api 1.1`] = `Object {
  "first_name": "Raphaël",
  "gender": "male",
  "id": 1,
  "last_name": "Benitte",
}`;

exports[`Snapshot testing on cli 1.1`] = `"test
"`;

exports[`Snapshot testing on files 1.1`] = `"id:         1
first_name: Raphaël
last_name:  Benitte
gender:     male
"`;

exports[`Snapshot testing on files 2.1`] = `"id:         1
first_name: Raphaël
last_name:  Benitte
gender:     male
"`;
